src/commands/queue.rs

```diff
@@ -1,25 +1,34 @@
-use std::cmp::{max, min};
+use std::{
+    cmp::{max, min},
+    ops::Add,
+    sync::Arc,
+    time::Duration,
+};
 
 use crate::{
-    strings::{NO_VOICE_CONNECTION, QUEUE_IS_EMPTY},
-    utils::{create_response, get_full_username, get_human_readable_timestamp},
+    client::GuildQueueInteractions,
+    events::modify_queue_handler::ModifyQueueHandler,
+    strings::{NO_VOICE_CONNECTION, QUEUE_EXPIRED, QUEUE_IS_EMPTY},
+    utils::{create_response, get_human_readable_timestamp},
 };
 use serenity::{
-    builder::CreateEmbed,
+    builder::{CreateButton, CreateComponents, CreateEmbed},
     client::Context,
     futures::StreamExt,
     model::{
-        channel::ReactionType,
+        channel::Message,
+        id::GuildId,
         interactions::{
             application_command::ApplicationCommandInteraction, message_component::ButtonStyle,
             InteractionResponseType,
         },
     },
-    prelude::SerenityError,
+    prelude::{RwLock, SerenityError, TypeMap},
 };
-use songbird::tracks::TrackHandle;
+use songbird::{tracks::TrackHandle, Event, TrackEvent};
 
 const EMBED_PAGE_SIZE: usize = 6;
+const COLLECTION_TIMEOUT: u64 = 3600;
```

Comment:
When this reaches 0 and it timeouts, does the embed disappear, like do we remove the embed message? If so, I would prefer `EMBED_TIMEOUT` and maybe with a little docstring specifying the time units (I know its seconds but still)

Reply:
The embed gets edited with `QUEUE_EXPIRED`, seen here:

https://github.com/aquelemiguel/parrot/blob/cd54d3432bf957022736a58007cf4bf20a9ea0d3/src/commands/queue.rs#L125-L135

Reply:
Then I would say just to use `EMBED_TIMEOUT`